docs/examples/plotting_functions/tricontourf.md

```diff
@@ -74,6 +74,101 @@ f
+f, ax, _ = tricontourf(x, y, z; boundary_nodes)
+scatter!(x, y, color = z, strokewidth = 1, strokecolor = :black)
+f
+```
+\end{examplefigure}
+
+Boundary nodes make it possible to support more complicated regions, possibly with holes, than is possible by only providing points themselves.
+
+\begin{examplefigure}{svg = true}
+```julia
+using CairoMakie, DelaunayTriangulation
+CairoMakie.activate!() # hide 
+
+## Start by defining the boundaries, and then convert to the appropriate interface 
+curve_1 = [
+    [(0.0, 0.0), (5.0, 0.0), (10.0, 0.0), (15.0, 0.0), (20.0, 0.0), (25.0, 0.0)],
+    [(25.0, 0.0), (25.0, 5.0), (25.0, 10.0), (25.0, 15.0), (25.0, 20.0), (25.0, 25.0)],
+    [(25.0, 25.0), (20.0, 25.0), (15.0, 25.0), (10.0, 25.0), (5.0, 25.0), (0.0, 25.0)],
+    [(0.0, 25.0), (0.0, 20.0), (0.0, 15.0), (0.0, 10.0), (0.0, 5.0), (0.0, 0.0)]
+] # outer-most boundary: counter-clockwise  
+curve_2 = [
+    [(4.0, 6.0), (4.0, 14.0), (4.0, 20.0), (18.0, 20.0), (20.0, 20.0)],
+    [(20.0, 20.0), (20.0, 16.0), (20.0, 12.0), (20.0, 8.0), (20.0, 4.0)],
+    [(20.0, 4.0), (16.0, 4.0), (12.0, 4.0), (8.0, 4.0), (4.0, 4.0), (4.0, 6.0)]
+] # inner boundary: clockwise 
+curve_3 = [
+    [(12.906, 10.912), (16.0, 12.0), (16.16, 14.46), (16.29, 17.06),
+    (13.13, 16.86), (8.92, 16.4), (8.8, 10.9), (12.906, 10.912)]
+] # this is inside curve_2, so it's counter-clockwise 
+curves = [curve_1, curve_2, curve_3]
+points = [
+    (3.0, 23.0), (9.0, 24.0), (9.2, 22.0), (14.8, 22.8), (16.0, 22.0),
+    (23.0, 23.0), (22.6, 19.0), (23.8, 17.8), (22.0, 14.0), (22.0, 11.0),
+    (24.0, 6.0), (23.0, 2.0), (19.0, 1.0), (16.0, 3.0), (10.0, 1.0), (11.0, 3.0),
+    (6.0, 2.0), (6.2, 3.0), (2.0, 3.0), (2.6, 6.2), (2.0, 8.0), (2.0, 11.0),
+    (5.0, 12.0), (2.0, 17.0), (3.0, 19.0), (6.0, 18.0), (6.5, 14.5),
+    (13.0, 19.0), (13.0, 12.0), (16.0, 8.0), (9.8, 8.0), (7.5, 6.0),
+    (12.0, 13.0), (19.0, 15.0)
+]
+boundary_nodes, points = convert_boundary_points_to_indices(curves; existing_points=points)
+edges = Set(((1, 19), (19, 12), (46, 4), (45, 12)))
+
+## Extract the x, y 
+x = getx.(points)
+y = gety.(points)
+z = (x .- 1) .* (y .+ 1) 
+f, ax, _ = tricontourf(x, y, z, boundary_nodes = boundary_nodes, edges = edges, levels = 30)
+f
+```
+\end{examplefigure}
+
+You can also provide a `Triangulation` object itself from DelaunayTriangulation.jl into `tricontourf`.
+
+\begin{examplefigure}{svg = true}
+```julia
+using CairoMakie, DelaunayTriangulation
+CairoMakie.activate!() # hide 
+
+using Random
+Random.seed!(1234)
+
+θ = [LinRange(0, 2π * (1 - 1/19), 20); 0]
+xy = Vector{Vector{Vector{NTuple{2,Float64}}}}()
+cx = [0.0, 3.0]
+for i in 1:2
+    push!(xy, [[(cx[i] + cos(θ), sin(θ)) for θ in θ]])
+    push!(xy, [[(cx[i] + 0.5cos(θ), 0.5sin(θ)) for θ in reverse(θ)]])
+end
+boundary_nodes, points = convert_boundary_points_to_indices(xy)
+tri = triangulate(points; boundary_nodes=boundary_nodes, check_arguments=false)
+z = [(x - 3/2)^2 + y^2 for (x, y) in each_point(tri)]
```

Comment:
this bit of the API I'm not sure about yet, is it clear for the user what the order of points is so that `z` matches it? Does the `tri` object retain the original order of points used? Or is the `each_point` function the source of truth

Reply:
The `each_point` iterator just iterates over `tri.points` or `eachcol(tri.points)` - order is preserved. I guess to be more explicit it could be something like

```julia
z = zeros(num_points(tri))
for i in each_point_index(tri) # or each_solid_vertex(tri)
    z[i] = (x[i] - 3/2)^2 + y[i]^2
end
```

What do you think? Or just a simple comment in the same line `# each_point preserves the indices of the points` I guess.